README
Dist::Zilla::Util::Test::KENTNL

This module is KENTNL's kit for testing Dist::Zilla.

Most of his modules should be moving to using the `dztest` model instead
which is more flexible source side.

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric <kentnl@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for {{$dist->name}}

{{$NEXT}}
 [Bugfix]
 - Update tooling to avoid test failures with '.' leaving @INC
 - Closes https://github.com/kentnl/Dist-Zilla-Util-Test-KENTNL/issues/3

 [Dependencies::Stats]
 - Dependencies changed since 1.005013, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑1)

1.005013 2015-10-19T11:29:36Z 0343510
 [Dependencies::Stats]
 - Dependencies changed since 1.005012, see misc/*.deps* for details
 - configure: +2
 - develop: +5 ↑1 -1 (suggests: ↑2)

 [Packaging]
 - Use a simpler version of DynamicPrereqs invokation, thanks @karenetheridge
 - Closes #1

1.005012 2015-04-13T03:25:53Z 2f733bb
 [Tests]
 - Todoize more of the "sniff out Moose loaders" test for now.

1.005011 2015-04-12T03:04:32Z 5d69b6a
 [Dependencies::Stats]
 - Dependencies changed since 1.005010, see misc/*.deps* for details
 - configure: ↓1
 - develop: +1
 - runtime: ↓1 -1
 - test: ↓1

 [Minor]
 - Drop utf8 and 5.008 minimums. Had intended to do this in previous release, but slipped my mind.
 - Some doc fixes too.
 - Avoid issues with older chainsmokes needing sanity

1.005010 2015-04-11T10:03:55Z 63f0907
 [Dependencies::Stats]
 - Dependencies changed since 1.005000, see misc/*.deps* for details
 - develop: +3 ↑3 -2 (recommends: ↓1, suggests: ↑2)
 - runtime: +2 -1 (recommends: +2 ↓1 -1)
 - test: (recommends: ↓1)

 [Minor]
 - Moved to Moo where Moose was required
 - Load Dzil lazier for App::Commands'

1.005000 2014-11-06T15:24:12Z 0194343
 [Dependencies::Stats]
 - Dependencies changed since 1.004002, see misc/*.deps* for details
 - develop: +1 -1 (recommends: ↑1)
 - runtime: +1 -1 (recommends: +1 ↑1)
 - test: (recommends: ↑1)

 [Features]
 - Now optionally uses Test::TempDir::Tiny for the temporary directory.

 [Internals]
 - use recommended.pm to load optional modules.

1.004002 2014-11-01T11:43:22Z dc767c8
 [Bugfix]
 - Require a version of Test::More with a useable subtest.

 [Dependencies::Stats]
 - Dependencies changed since 1.004001, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: ↑1
 - test: ↑1

1.004001 2014-10-30T12:56:17Z cab1d68
 [Bugfix]
 - Soften DPath dependency till upstream can fix it for 5.8

 [Dependencies::Stats]
 - Dependencies changed since 1.004000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +4 -1 (recommends: +1, suggests: ↑2)
 - runtime: -1 (recommends: +1 ↑1)
 - test: +1 (recommends: ↑2)

1.004000 2014-10-12T15:44:27Z b3d5941
 [Dependencies::Stats]
 - Dependencies changed since 1.003003, see misc/*.deps* for details
 - develop: +3 ↑1 (suggests: ↑1)
 - runtime: +1
 - test: +1

 [Features]
 - dztest now has ->run_command() proxy for DZ:App::Tester::test_dzil

1.003003 2014-09-28T07:27:06Z
 [Bugfix]
 - spew_raw to avoid \r\n translation on Win32

 [Dependencies::Stats]
 - Dependencies changed since 1.003002, see misc/*.deps* for details
 - develop: (suggests: ↑1)

1.003002 2014-09-07T09:07:01Z
 [00 Bugfix]
 - Warn and auto-wrap $expected in meta_path_deeply to avoid other instances of
   kentnl/Dist-Zilla-Plugin-Prereqs-MatchInstalled#3

 [Dependencies::Stats]
 - Dependencies changed since 1.003001, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: (recommends: ↑1)
 - test: (recommends: ↑1)

1.003001 2014-08-29T09:27:43Z
 [00 Bugfix]
 - Make sure failing to create a build dir is handled properly

 [Dependencies::Stats]
 - Dependencies changed since 1.003000, see misc/*.deps* for details
 - runtime: ↑1

1.003000 2014-08-29T07:46:22Z
 [00 Minor]
 - Add create_plugin utility.
 - Test enhancements.

 [Dependencies::Stats]
 - Dependencies changed since 1.002000, see misc/*.deps* for details
 - runtime: +2

 [Tests]
 - Test coverage improved.

1.002000 2014-08-22T17:40:28Z
 [00 Minor]
 - Prefer Test::Differences everywhere it is possible.

 [Dependencies::Stats]
 - Dependencies changed since 1.001002, see misc/*.deps* for details
 - develop: +1
 - runtime: (recommends: +1)

1.001002 2014-08-21T05:38:29Z
 [Bugfix]
 - make sure build is in fact, always invoked when builder is required.

 [Dependencies::Stats]
 - Dependencies changed since 1.001001, see misc/*.deps* for details
 - develop: (suggests: ↑1)

1.001001 2014-08-18T23:05:17Z
 [Bugfix]
 - Not all versions of Data::DPath exist on <5.10, so if you happen to say "Ignore that" somehow the dpath related
   tests will instead TODOskip

 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - runtime: +1

1.001000 2014-08-18T21:14:28Z
 [00 Major]
 - add `dztest` function.
 - add `::dztest` package which has been floating around in various dists so far.

 [Dependencies::Stats]
 - Dependencies changed since 1.000003, see misc/*.deps* for details
 - develop: +2 ↑1 (suggests: ↑2)
 - runtime: +5 (recommends: +1)
 - test: -1

1.000003 2014-08-03T12:37:40Z
 [00 Trivial]
 - Migrate github links to org ( was advised in last release ... was wrong -_- )

 [Dependencies::Stats]
 - Dependencies changed since 1.000002, see misc/*.deps* for details
 - configure: ↓1
 - develop: (suggests: ↑1)
 - test: -4 (recommends: ↑1 -2)

1.000002 2014-07-25T06:22:28Z
 [00 Trivial]
 - Newer report versions script autoadded.

 [Dependencies::Stats]
 - Dependencies changed since 1.000001, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - test: +2 ↑1 (recommends: +1)

1.000001 2014-07-13T18:11:06Z
 [00 Trivial]
 - Minor packaging changes.

 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - configure: (recommends: ↑1)
 - develop: +3 -3 (suggests: +2 -1)
 - test: (recommends: ↑2)

1.000000 2014-02-02T07:19:39Z
 [00 Major]
 - Toolkit switched to EUMM
 - Toolkit deps softened to recommendations.
 - Version moved to mantissa = 6 ( primary reason for 0 → 1 )

 [Dependencies::Aggregate]
 - 5.008 : was 197 → 199 :(
 - 5.010 : was 178 → 184 :(
 - 5.019 : was 161 → 164 :(
 - Growth induced in dependencies.
 - Limited by DZIL → Path::Class → Module::Build

 [Dependencies::Stats]

Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.039.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "KENTNL's DZil plugin testing tool",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "CPAN::Meta::Requirements" => "2.120620",
    "ExtUtils::MakeMaker" => 0,
    "Module::Metadata" => 0
  },
  "DISTNAME" => "Dist-Zilla-Util-Test-KENTNL",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Util::Test::KENTNL",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Data::DPath" => 0,
    "Dist::Zilla::App::Tester" => 0,
    "Dist::Zilla::Tester" => 0,
    "Dist::Zilla::Util" => 0,
    "Module::Runtime" => 0,
    "Moo" => 0,
    "Path::Tiny" => "0.050",
    "Sub::Exporter" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.96",
    "Try::Tiny" => 0,
    "recommended" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::App::Command::authordeps" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "1.005014",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Data::DPath" => 0,
  "Dist::Zilla::App::Command::authordeps" => 0,
  "Dist::Zilla::App::Tester" => 0,
  "Dist::Zilla::Tester" => 0,
  "Dist::Zilla::Util" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Runtime" => 0,
  "Moo" => 0,
  "Path::Tiny" => "0.050",
  "Sub::Exporter" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.96",
  "Try::Tiny" => 0,
  "recommended" => 0,
  "strict" => 0,
  "warnings" => 0
);

# inserted by Dist::Zilla::Plugin::DynamicPrereqs 0.016
# Avoid issue with sanity bugs and Moose penalties
my $module = "Dist::Zilla::App::Command::chainsmoke";
requires($module, '1.15') if has_module($module);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

# inserted by Dist::Zilla::Plugin::DynamicPrereqs 0.016
sub has_module {
  my ($module, $version_or_range) = @_;
  require Module::Metadata;
  my $mmd = Module::Metadata->new_from_module($module);
  return undef if not $mmd;
  return $mmd->version($module) if not defined $version_or_range;

  require CPAN::Meta::Requirements;
  my $req = CPAN::Meta::Requirements->new;
  $req->add_string_requirement($module => $version_or_range);
  return 1 if $req->accepts_module($module => $mmd->version($module));
  return 0;
}

sub requires {
  goto &runtime_requires;
}

sub runtime_requires {
  my ($module, $version_or_range) = @_;
  $version_or_range = '0' if not $version_or_range;
  $WriteMakefileArgs{PREREQ_PM}{$module} = $FallbackPrereqs{$module} = $version_or_range;
  return;
}


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    KENTNL DZil
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');

#$bundle->remove_policy('CodeLayout::RequireUseUTF8');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.012.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4204"
  },
  "configure_requires" => {
    "Module::Build" => "0.4204"
  },
  "dist_abstract" => "KENTNL's DZil plugin testing tool",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Util-Test-KENTNL",
  "dist_version" => "1.000000",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Util::Test::KENTNL",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Dist::Zilla::Tester" => 0,
    "Sub::Exporter" => 0,
    "Test::DZil" => 0,
    "Try::Tiny" => 0,
    "perl" => "5.008",
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "1.001002",
    "version" => 0
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4204",
  "Test::Fatal" => 0,
  "Test::More" => "1.001002",
  "version" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
